### PR TITLE
[scraperhelper] Deprecate ScraperControllerSettings, use ControllerConfig instead

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Deprecate ScraperControllerSettings, use ~Scraper~ControllerConfig instead

**Link to tracking Issue:**
#6767

### CODE REVIEW

File: receiver/scraperhelper/settings.go
Comment (TylerHelmuth) [inline receiver/scraperhelper/settings.go:1]: We should rename this file

Comment (atoulme) [reply receiver/scraperhelper/settings.go:1]: Sure, done.

File: receiver/scraperhelper/config.go [hunk 1]
```diff
@@ -20,7 +20,13 @@ var (
...
 // configuration. Scraper controller receivers can embed this struct, instead
 // of receiver.Settings, and extend it with more fields if needed.
-type ScraperControllerSettings struct {
+// Deprecated: [v0.94.0] Use ScraperControllerConfig instead
+type ScraperControllerSettings = ScraperControllerConfig
+
+// ScraperControllerConfig defines common settings for a scraper controller
+// configuration. Scraper controller receivers can embed this struct, instead
+// of receiver.Settings, and extend it with more fields if needed.
+type ScraperControllerConfig struct {
```
Comment (bogdandrutu) [inline receiver/scraperhelper/config.go:29]: remove `Scrapper` prefix, same reason as `configgrpc`

**Review by bogdandrutu — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by bogdandrutu** [comment]
> However, the parameter component.Type is not used. Should we remove it?

Please

**Comment by atoulme** [comment]
Done, it's removed.

File: receiver/scraperhelper/config.go [hunk 2]
```diff
@@ -34,15 +40,26 @@ type ScraperControllerSettings struct {
...
 // settings with a collection interval of one minute.
-func NewDefaultScraperControllerSettings(component.Type) ScraperControllerSettings {
-	return ScraperControllerSettings{
+// Deprecated: [v0.94.0] Use NewDefaultControllerConfig instead
+func NewDefaultScraperControllerSettings(component.Type) ControllerConfig {
+	return ControllerConfig{
+		CollectionInterval: time.Minute,
+		InitialDelay:       time.Second,
+		Timeout:            0,
+	}
```
Comment (bogdandrutu) [inline receiver/scraperhelper/config.go:49]:
```suggestion
	return NewDefaultControllerConfig()
```

File: receiver/scraperhelper/config.go [hunk 3]
```diff
@@ -20,8 +20,14 @@ var (
 // ScraperControllerSettings defines common settings for a scraper controller
 // configuration. Scraper controller receivers can embed this struct, instead
 // of receiver.Settings, and extend it with more fields if needed.
-type ScraperControllerSettings struct {
-	// CollectionInterval sets the how frequently the scraper
+// Deprecated: [v0.94.0] Use ControllerConfig instead
```
Comment (bogdandrutu) [inline receiver/scraperhelper/config.go:23]: Update version.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**